src/main/networking/mac-ca.ts

```diff
@@ -0,0 +1,159 @@
+    // as it wouldn't include _client_ certificates.
+    const rootCerts = '/System/Library/Keychains/SystemRootCertificates.keychain';
+
+    await fs.promises.access(rootCerts, fs.constants.R_OK);
+    yield rootCerts;
+  } catch (ex) { /* swallow the error */ }
+}
+
+/**
+  * Asynchronously enumerate PEM-encoded certificates from the given keychain in
+  * undefined order.
+  *
+  * @param workdir A temporary directory where files can be written.
+  * @param keychain The full path to the keychain database to enumerate.
+  */
+async function* getFilteredCertificates(workdir: string, keychain: string): AsyncIterable<string> {
+  console.debug(`getting certificates from ${ keychain }...`);
+
+  const certIterator = getPEMCertificates(workdir, keychain);
+
+  for await (const certPEM of certIterator) {
+    const cert = new crypto.X509Certificate(certPEM);
+    const certPath = path.join(workdir, 'cert.pem');
+
+    if (!cert.ca) {
+      console.debug('Skipping non-CA certificate', cert.subject);
+      continue;
+    }
+    await fs.promises.writeFile(certPath, certPEM, 'utf-8');
+    try {
+      await spawnFile('/usr/bin/security', ['verify-cert', `-c${ certPath }`, '-L', '-l', '-Roffline'], { stdio: console });
+    } catch (ex) {
+      console.debug('Skipping untrusted certificate', cert.subject);
+      continue;
+    }
+    yield certPEM;
+  }
+}
+
+/**
+ * Enumerate all system certificates as PEM, in undefined order.  This does not
+ * do the necessary processing to ensure they are valid for our use.
+ *
+ * @param workdir A temporary directory where files can be written.
+ * @param keychain Optional absolute path to a specific Keychain database to use.
+ */
+async function* getPEMCertificates(workdir: string, keychain?: string): AsyncIterable<string> {
+  // In order to avoid issues on machine with a very large number of certificates,
+  // write all certificates (in PEM format) to a file, and then read that file out.
+  const pemMarker = '-----END CERTIFICATE-----';
+  const pemFilePath = path.join(workdir, 'all-certs.pem');
+  const pemFile = await fs.promises.open(pemFilePath, fs.constants.O_CREAT | fs.constants.O_TRUNC | fs.constants.O_WRONLY);
+  const pemFileStream = fs.createWriteStream(pemFilePath, { fd: pemFile });
+  const args = ['find-certificate', '-a', '-p'];
+
+  if (keychain) {
+    args.push(keychain);
+  }
+  await spawnFile('/usr/bin/security', args, { stdio: ['ignore', pemFileStream, console] });
+  await util.promisify((cb: (err?: Error | null | undefined) => void) => pemFileStream.close(cb))();
+  await pemFile.close();
+
+  let pemLines: string[] = [];
+
+  for await (const line of readFileByLine(pemFilePath)) {
+    pemLines.push(line);
+    if (line === pemMarker) {
+      yield pemLines.join('\n');
+      pemLines = [];
+    }
+  }
+
+  if (pemLines.length > 0 && pemLines[pemLines.length - 1] === pemMarker) {
+    yield pemLines.join('\n');
+  }
+}
+
+/**
```

Comment:
Ah; I saw references to that (without the context) and thought they were referring to [`readline`](https://nodejs.org/docs/latest-v16.x/api/readline.html) — which looked like bindings to [GNU `readline`](https://tiswww.case.edu/php/chet/readline/rltop.html).

[`linebyline`](https://www.npmjs.com/package/linebyline) (the new name for the library to avoid the name conflict) does look better, though I'm not really a fan of pulling in more dependencies for things we can relatively easily replace with vendored code…

Reply:
@mook-as I'd like to echo your sentiment; if the maintenance burden isn't too great, I think it makes sense to limit our dependencies.